src/rdkafka_admin.c

```diff
@@ -6666,3 +6773,705 @@ rd_kafka_DescribeConsumerGroups_result_groups(
+        rd_kafka_ClusterDescription_destroy(ptr);
+}
+/**
+ * @brief Send DescribeClusterRequest. Admin worker compatible callback.
+ */
+static rd_kafka_resp_err_t
+rd_kafka_admin_DescribeClusterRequest(rd_kafka_broker_t *rkb,
+                                      const rd_list_t *topics /*(char*)*/,
+                                      rd_kafka_AdminOptions_t *options,
+                                      char *errstr,
+                                      size_t errstr_size,
+                                      rd_kafka_replyq_t replyq,
+                                      rd_kafka_resp_cb_t *resp_cb,
+                                      void *opaque) {
+        rd_kafka_resp_err_t err;
+        int include_cluster_authorized_operations;
+
+        include_cluster_authorized_operations =
+            rd_kafka_confval_get_int(&options->include_authorized_operations);
+
+        err = rd_kafka_MetadataRequest(
+            rkb, NULL, "describe cluster", rd_false /*no auto create*/,
+            include_cluster_authorized_operations,
+            rd_false /*!include topic authorized operations */,
+            rd_false /*cgrp update*/, rd_false /* force_rack */, NULL, resp_cb,
+            1 /* force */, opaque);
+
+        if (err) {
+                rd_snprintf(errstr, errstr_size, "%s", rd_kafka_err2str(err));
+                return err;
+        }
+
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+}
+
+/**
+ * @brief Parse DescribeCluster and create ADMIN_RESULT op.
+ */
+static rd_kafka_resp_err_t
+rd_kafka_DescribeClusterResponse_parse(rd_kafka_op_t *rko_req,
+                                       rd_kafka_op_t **rko_resultp,
+                                       rd_kafka_buf_t *reply,
+                                       char *errstr,
+                                       size_t errstr_size) {
+        rd_kafka_metadata_internal_t *mdi = NULL;
+        rd_kafka_resp_err_t err;
+        rd_kafka_ClusterDescription_t *clusterdesc = NULL;
+        rd_list_t topics       = rko_req->rko_u.admin_request.args;
+        rd_kafka_broker_t *rkb = reply->rkbuf_rkb;
+        int32_t cluster_authorized_operations;
+        char *cluster_id = NULL;
+        int controller_id;
+        rd_kafka_op_t *rko_result = NULL;
+
+        err        = rd_kafka_parse_Metadata(rkb, NULL, reply, &mdi, &topics);
+        cluster_id = mdi->cluster_id;
+        controller_id                 = mdi->controller_id;
+        cluster_authorized_operations = mdi->cluster_authorized_operations;
+        if (err)
+                goto err;
+        rko_result = rd_kafka_admin_result_new(rko_req);
+        rd_list_init(&rko_result->rko_u.admin_result.results, 1,
+                     rd_kafka_ClusterDescription_free);
+
+        rd_list_t *authorized_operations;
+        authorized_operations =
+            rd_kafka_AuthorizedOperations_parse(cluster_authorized_operations);
+
+        clusterdesc = rd_kafka_ClusterDescription_new(
```

Comment:
I think we should have a function `rd_kafka_ClusterDescription_new(rd_kafka_metadata_internal_t  *mdi)`. The current function can be `new0` but not with list of nodes instead of metadata.

Reply:
Since it's an internal function used only in one place, I am changing it to just accept rd_kafka_metadata_internal_t.